### PR TITLE
Update Visibility docs on maintainSize

### LINKED ISSUE #53073: Visibility docs on maintainSize imply there are other possible configurations when there are not
`Visibility` `maintainSize` docs have a big paragraph explaining the costs of `maintainSize` with `maintainAnimation` and `maintainState`. The whole language looks like a good tutorial to know when do we choose to apply each one of those options. But that's not the case as we don't really have any options here, `maintainSize = true` forces us to use `maintainAnimation = true` and `maintainState = true`.

To be clearer. the docs only say that `To set this, [maintainAnimation] must also be set.`, so if `maintainSize` is set to `true`, then `maintainAnimation` should also be `true`. But on `maintainAnimation` docs we see that `To set this, [maintainState] must also be set.`, so implicitly **if `maintainSize` is set to `true`, then both `maintainAnimation` and `maintainState` should also be `true`.**

`maintainSize` docs should include that `maintainState` should also be true, and the big paragraph explaining it's costs should make it clearer that it's just an explanation on the costs and not something we can actually choose when to use each option separately.

### PR DESCRIPTION
**Pull request by younghwan**
## Description

To make the documentation clearer, I added a statement about maintainState

## Related Issues

Fixes #53073

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/visibility.dart [hunk 1]
```diff
@@ -173,6 +173,9 @@ class Visibility extends StatelessWidget {
   ///
   /// If this property is true, [Opacity] is used instead of [Offstage].
   ///
+  /// If this property is true, then both [maintainAnimation] and
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/visibility.dart:176]: This seems to just be a duplication of the sentence in line 164? Remove one of them?

Comment (younghwan) [reply packages/flutter/lib/src/widgets/visibility.dart]: Thanks for the advice. I will do that.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by feinstein** [comment]
This doesn't completely fixes the issue I opened, it just adds one of the recommendations I gave.

There's a also the problem with:

```dart
  /// Maintaining the size when the widget is not [visible] is not notably more
  /// expensive than just keeping animations running without maintaining the
  /// size, and may in some circumstances be slightly cheaper if the subtree is
  /// simple and the [visible] property is frequently toggled, since it avoids
  /// triggering a layout change when the [visible] property is toggled. If the
  /// [child] subtree is not trivial then it is significantly cheaper to not
  /// even keep the state (see [maintainState]).
```

Which goes in depth to explain the costs of maintaining the size, animations and state, but we don't actually have an option as if we select to `mantainSize` we are obligated to also add `maintainAnimations` and `mantainState`.

That's the part that I think needs clarification. It looks like something that was written a while ago, but the Framework evolved and added constraints that the docs don't fit anymore.

**Pull request merged**